main.js

```diff
@@ -0,0 +1,258 @@
+  let targetCurrency = document.getElementById('targetCurrency').value;
+  let amount = parseFloat(document.getElementById('amount').value);
+
+  //Validate the amount
+  if (isNaN(amount)){
+    document.getElementById('result').innerText = 'Please enter a valid amount';
+    return;
+  }
+
+  //Find the index of the main currency in the currencyData array
+  const mainCurrencyIndex = currencyData.findIndex(currency => currency.baseCurrency === mainCurrency);
+
+  if (mainCurrencyIndex !== -1) {
+    //Check if the target currency is in the rates for the main currency
+    if (currencyData[mainCurrencyIndex].rates.hasOwnProperty(targetCurrency)) {
+      const conversionRate = currencyData[mainCurrencyIndex].rates[targetCurrency];
+      const result = amount * conversionRate;
+
+      document.getElementById('result').innerText = `${amount} ${mainCurrency} equals ${result.toFixed(2)} ${targetCurrency}`;
+
+      return result;
+    } else {
+      document.getElementById('result').innerText = 'Conversion rate not available for the target currency';
+    }
+  } else {
+    document.getElementById('result').innerText = 'Main currency not found';
+  }
+}
+
+//Search Function Week2
+function searchCurrency() {
+    const searchFrom = document.getElementById('searchFrom').value;
+    const searchResultContainer = document.getElementById('searchResult');
+
+    //console.log('Search from:', searchFrom);
+
+    //Find the base currency in the currencyData array
+    const baseCurrency = currencyData.find(currency => currency.baseCurrency === searchFrom);
+
+    if (baseCurrency) {
+        const rates = baseCurrency.rates;
+
+        //console.log('Base currency found:', baseCurrency);
+
+        //Clear previous search results
+        if (rates && Object.keys(rates).length > 0) {
+            //Clear previous search results
+            searchResultContainer.innerHTML = '';
+
+            //Loop through rates and display results
+            for (const currencyCode in rates) {
+                if (rates.hasOwnProperty(currencyCode)) {
+                    const rate = rates[currencyCode];
+
+                    //Create HTML elements
+                    const resultItem = document.createElement('p');
+                    resultItem.textContent = `Rate from ${searchFrom} to ${currencyCode}: ${rate}`;
+
+                    //Append to the container
+                    searchResultContainer.appendChild(resultItem);
+                }
+            }
+        } else {
+            //console.log('Currency rates not found or empty.');
+            searchResultContainer.innerText = 'Currency rates not found.';
+        }
+    } else {
+        //console.log('Base currency not found.');
+        searchResultContainer.innerText = 'Base currency not found.';
+    }
+}
+
+
+// Week3 Special Value Alert
+const notification = document.getElementById('notification')
+
+function checkCurrencyRate() {
```

Comment:
Right now this function serves only to check whether the currency exists right? For this purpose, we can use the already existing table

Reply:
No, it is a currency rate alarm. Not a proper alarm as my data is not real-time.

Reply:
but right now it checks only wheter there is some rate, but it should rather check if the rate extends some value e.g. 
USD/DKK hot rate is 2

In case
USD =＞ DKK 0.5 // no alarm
in case
USD => DKK 2.1 // alarm

Reply:
Oh yes that is true 🙀